dso-l2/src/main/java/com/tc/objectserver/entity/ActiveToPassiveReplication.java

```diff
@@ -263,7 +277,9 @@ public void removePassive(NodeID nodeID) {
 //  a double ack locally but that is ok.  acknowledge is loose and can tolerate it. 
     if (activated) {
 //  remove the passive node from the sender first.  nothing else is going out
-      this.replicationSender.removePassive(nodeID);
+      BarrierCompletion barrierCompletion = new BarrierCompletion();
+      this.replicationSenderMessageSink.addSingleThreaded(new RemovePassiveMessage(nodeID, () -> barrierCompletion.complete()));
+      barrierCompletion.waitForCompletion();
       removeWaiters(nodeID);
```

Comment:
Can passives be added and removed via synchronization?  They don't happen often and if we can remove the extra step and be more straightforward, I think it is better.